static/app/views/starfish/views/screens/screenLoadSpans/eventSamplesTable.tsx

```diff
@@ -0,0 +1,230 @@
+import {Fragment} from 'react';
+import {browserHistory} from 'react-router';
+import styled from '@emotion/styled';
+
+import {LinkButton} from 'sentry/components/button';
+import GridEditable, {GridColumnHeader} from 'sentry/components/gridEditable';
+import SortLink from 'sentry/components/gridEditable/sortLink';
+import Link from 'sentry/components/links/link';
+import Pagination, {CursorHandler} from 'sentry/components/pagination';
+import {Tooltip} from 'sentry/components/tooltip';
+import {IconProfiling} from 'sentry/icons/iconProfiling';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
+import {TableData, TableDataRow} from 'sentry/utils/discover/discoverQuery';
+import EventView, {isFieldSortable, MetaType} from 'sentry/utils/discover/eventView';
+import {getFieldRenderer} from 'sentry/utils/discover/fieldRenderers';
+import {fieldAlignment, Sort} from 'sentry/utils/discover/fields';
+import {generateProfileFlamechartRoute} from 'sentry/utils/profiling/routes';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {TableColumn} from 'sentry/views/discover/table/types';
+import {DeviceClassSelector} from 'sentry/views/starfish/views/screens/screenLoadSpans/deviceClassSelector';
+
+type Props = {
+  columnNameMap: Record<string, string>;
+  cursorName: string;
+  eventIdKey: string;
+  eventView: EventView;
+  isLoading: boolean;
+  profileIdKey: string;
+  sort: Sort;
+  sortKey: string;
+  data?: TableData;
+  pageLinks?: string;
+  showDeviceClassSelector?: boolean;
+};
+
+const ICON_FIELDS = ['profile.id', 'profile_id'];
```

Comment:
nit: I feel like it might be cleaner to restrict `eventIdKey` and `profileIdKey` to just the expected string values, ie `"id" | "transaction.id"` and `'profile.id'|'profile_id'`. Especially since `ICON_FIELDS` already expects those profile id values.

Or maybe an even better solution is to get rid of these props, and expect `data` to be transformed to the correct format.

Reply:
I added the specific values, I don't really want to massage the data too much from the events result yet, I feel like those transformations make it harder to follow data to the components and this is kind of small footprint 😬